tests/functional/object/mcg/test_noobaa_bucket_error_state_mode.py

```
@@ -0,0 +1,560 @@
+
+    try:
+        for mode in TimeoutSampler(
+            timeout=BUCKET_MODE_TIMEOUT,
+            sleep=BUCKET_MODE_POLL_INTERVAL,
+            func=_get_bucket_mode,
+            mcg_obj=mcg_obj,
+            bucket_name=bucket_name,
+        ):
+            last_mode = mode
+            if check_not_optimal and mode != "OPTIMAL":
+                log.info(f"Bucket {bucket_name} entered non-OPTIMAL mode: {mode}")
```

Ruff (0.15.15)

[warning] Logging statement uses f-string

(G004)